brave/brave-core
# [ads] Only collect, store and redeem essential metadata

Resolves https://github.com/brave/brave-browser/issues/38955



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

### Non-Brave Rewards user:

_Test Case #1:_
- Fresh install
- Navigate to search.brave.software
- Search for `giraffe`
**EXPECT RESULT:** Ad viewed event should not occur

- Navigate to https://www.brave.com/vertical-tabs/ to trigger a conversion
**EXPECT RESULT:** View-through conversion should not occur

- The following state should be verified:
    - `Default/ads_service/client.json` state should be empty
    - `Default/ads_service/confirmations.json` state should be empty
    - `Default/ads_service/database.sqlite`:
        - All tables should be empty

--

_Test Case #2:_
- Fresh install
- Navigate to search.brave.software
- Search for `giraffe`
- Click `[Staging] Brave Ads` ad
**EXPECT RESULT:** `Clicked search result ad with placement id # and creative instance id #` and `Successfully saved creative set conversions` appear in the console log

- Navigate to https://www.brave.com/vertical-tabs/ to trigger a conversion
**EXPECT RESULT:** `Successfully processed conversion confirmation for search_result_ad with transaction id # and creative instance id #` appears in the console log and `{"conversion":[{"action":"click"}],"creativeInstanceId":"#","transactionId":"#","type":"conversion"}` should be sent to
https://search.anonymous.ads.bravesoftware.com/v3/confirmation/#

- The following state should be verified:
    - `Default/ads_service/client.json` state should be empty
    - `Default/ads_service/confirmations.json` state should be empty
    - `Default/ads_service/database.sqlite`:
        - The `creative_set_conversions` table should contain a list of conversions for the associated creative set
        - The `ad_events` table should only contain click and conversion events
        - All other tables should be empty

--

_Test Case #3:_
- Regression test Brave News ads

### Brave Rewards user:

_Test Case #1:_
- Regression test search result ads and Brave News ads
